### PR TITLE
docs: Add Dex IDP setup for OIDC

### PR DESCRIPTION
**Pull request by yolossn**
This patch adds documentation for setting up
Dex with Github connection as OIDC provider.

Signed-off-by: Santhosh Nagaraj S <santhosh@kinvolk.io>

### CODE REVIEW

File: docs/authorization.md [hunk 1]
```diff
@@ -153,6 +153,102 @@ Now the rule to add the roles to the token is setup, the roles will be available
   <img width="100%" src="./images/auth0-token.gif">
 </p>
 
+# Preparing Dex with github connector as OIDC provider for Nebraska.
+
+## Setting up Github App to be used as connector for Dex.
+
+- Create a New Organization in Github.
```
Comment (joaquimrocha) [inline docs/authorization.md:160]: Don't capitalize New Organization. If you want to highlight, use bold or link to how it's done in the GH docs.

File: docs/authorization.md [hunk 2]
```diff
@@ -153,6 +153,102 @@ Now the rule to add the roles to the token is setup, the roles will be available
...
+
+- Now you need a github app, go to `https://github.com/organizations/<ORG>/settings/apps/new` and fill
```
Comment (joaquimrocha) [inline docs/authorization.md:162]: Capitalize Github in order to be consistent with the rest.

File: docs/authorization.md [hunk 3]
```diff
@@ -153,6 +153,102 @@ Now the rule to add the roles to the token is setup, the roles will be available
...
+  the following fields:
+
+  - `GitHub App name` - just put some fancy name, must be unique I
+    think
```
Comment (joaquimrocha) [inline docs/authorization.md:166]: Don't add the `must be unique I think`. Users will figure out if it needs to be unique, or we can check.

File: docs/authorization.md [hunk 4]
```diff
@@ -153,6 +153,102 @@ Now the rule to add the roles to the token is setup, the roles will be available
...
+
+  - `Homepage URL` - `http://localhost:8000`
```
Comment (joaquimrocha) [inline docs/authorization.md:168]: This is for a testing environment, right? So maybe we have to add some disclaimer about it in the section's intro.

File: docs/authorization.md [hunk 5]
```diff
@@ -153,6 +153,102 @@ Now the rule to add the roles to the token is setup, the roles will be available
...
+
+  - Choose `Install App` on the left of the page and perform the
+    installation
+
+## Creating Github Teams.
+
+- Create two teams in your organization with the following names
+  `admin` and `viewer`.
+
+- Add the admin users to both `admin` and `viewer` team. Add the member users to
```
Comment (joaquimrocha) [inline docs/authorization.md:200]: member users -> non-admin users

File: docs/authorization.md [hunk 6]
```diff
@@ -153,6 +153,102 @@ Now the rule to add the roles to the token is setup, the roles will be available
...
+    loadAllGroups: true
+    teamNameField: slug
+    useLoginAsID: true
+    
+enablePasswordDB: true
+```
+
+- Run Dex using the example configuration.
+
+> ./bin/dex serve example.yaml
```
Comment (joaquimrocha) [inline docs/authorization.md:246]: Should this be done through a container image instead?

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Left a few comments.

File: docs/authorization.md (hunk 4 above)
Comment (yolossn) [reply docs/authorization.md:167]: All the OIDC configurations docs are for running local, I think the changes required for hosting are relatively straightforward.

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**